internal/build/helper.go

```diff
@@ -0,0 +1,187 @@
+package build
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/buildpacks/imgutil/layout/sparse"
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/cmd"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/daemon"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/buildpacks/pack/pkg/logging"
+)
+
+const (
+	DockerfileKindBuild = "build"
+	DockerfileKindRun   = "run"
+)
+
+type Extensions struct {
+	Extensions []buildpack.GroupElement
+}
+
+func (extensions *Extensions) DockerFiles(kind string, path string, logger logging.Logger) ([]buildpack.DockerfileInfo, error) {
+	var dockerfiles []buildpack.DockerfileInfo
+	for _, ext := range extensions.Extensions {
+		dockerfile, err := extensions.ReadDockerFile(path, kind, ext.ID)
+		if err != nil {
+			return nil, err
+		}
+		if dockerfile != nil {
+			logger.Debugf("Found %s Dockerfile for extension '%s'", kind, ext.ID)
+			switch kind {
+			case DockerfileKindBuild:
+				// will implement later
+			case DockerfileKindRun:
+				buildpack.ValidateRunDockerfile(dockerfile, logger)
+			default:
+				return nil, fmt.Errorf("unknown dockerfile kind: %s", kind)
+			}
+			dockerfiles = append(dockerfiles, *dockerfile)
+		}
+	}
+	return dockerfiles, nil
+}
+
+func (extensions *Extensions) ReadDockerFile(path string, kind string, extID string) (*buildpack.DockerfileInfo, error) {
+	dockerfilePath := filepath.Join(path, kind, escapeID(extID), "Dockerfile")
+	if _, err := os.Stat(dockerfilePath); err != nil {
+		return nil, nil
+	}
+	return &buildpack.DockerfileInfo{
+		ExtensionID: extID,
+		Kind:        kind,
+		Path:        dockerfilePath,
+	}, nil
+}
```

Comment:
Could we make `func (e *Extender) dockerfileFor(kind, extID string)` in lifecycle/extender.go a static and public method to avoid duplicating this logic?